File: python/bob/bob.py
def response(hey_bob):
    message = hey_bob.strip()
    
    is_yelling = message == message.upper() and message != message.lower()
    is_question = message[-1:] == '?'
    is_nothing = len(message) == 0
    if is_question and is_yelling:
        return 'Calm down, I know what I\'m doing!'
    elif is_question:
        return 'Sure.'
    elif is_yelling:
        return 'Whoa, chill out!'
    elif is_nothing:
        return 'Fine. Be that way!'
    else:
        return 'Whatever.'

File: csharp/bob/Bob.cs
using System.Linq;

public static class Bob
{
    public static string Response(string statement)
    {
        statement = statement.Trim();
        bool isQuestion = statement.EndsWith('?');
        bool hasLetters = statement.Where(c => Char.IsLetter(c)).Count() > 0;
        bool isYelling = hasLetters && statement.Equals(statement.ToUpperInvariant());
        if (statement.Length == 0)
        {
            return "Fine. Be that way!";
        }
        else if (isYelling && isQuestion)
        {
            return "Calm down, I know what I'm doing!";
        }
        else if (isYelling)
        {
            return "Whoa, chill out!";
        }
        else if (isQuestion)
        {
            return "Sure.";
        }

        return "Whatever.";
    }
}